internal/provider/webhook/resource_notifications_webhook.go

```diff
@@ -0,0 +1,545 @@
+The destination must be able to use the HCP webhook 
+[payload](https://developer.hashicorp.com/hcp/docs/hcp/admin/projects/webhooks#webhook-payload).`,
+						Validators: []validator.String{
+							hcpvalidator.URL(),
+						},
+					},
+					"hmac_key": schema.StringAttribute{
+						Optional: true,
+						Description: "The arbitrary secret that HCP uses to sign all its webhook requests. This is a " +
+							"write-only field, it is written once and not visible thereafter.",
+						Sensitive: true,
+					},
+				},
+			},
+
+			// Optional fields
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the project to create the webhook under. " +
+					"If unspecified, the webhook will be created in the project the provider is configured with.",
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+
+			"description": schema.StringAttribute{
+				Optional:    true,
+				Description: "The webhook's description. Descriptions are useful for helping others understand the purpose of the webhook.",
+			},
+
+			"enabled": schema.BoolAttribute{
+				Optional:            true,
+				Computed:            true,
+				MarkdownDescription: "Indicates if the webhook should receive payloads for the subscribed events.",
+				Default:             booldefault.StaticBool(true),
+			},
+
+			"subscriptions": schema.ListNestedAttribute{
+				Optional:    true,
+				Description: "Set of events to subscribe the webhook to all resources or a specific resource in the project.",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"resource_id": schema.StringAttribute{
+							Optional: true,
+							Description: "Refers to the resource the webhook is subscribed to. " +
+								"If not set, the webhook subscribes to the emitted events listed in events for " +
+								"any resource in the webhook's project.",
+							Validators: nil,
+						},
+						"events": schema.ListNestedAttribute{
+							Required: true,
+							Description: "The information about the events of a webhook subscription. " +
+								"The service that owns the resource is responsible for maintaining events. " +
+								"Refer to the service's webhook documentation for more information.",
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"actions": schema.ListAttribute{
+										ElementType: types.StringType,
+										Required:    true,
+										Description: "The list of event actions subscribed for the resource type set as the [source](#source). " +
+											"For example, `[\"create\", \"update\"]`. " +
+											"When the action is '*', it means that the webhook is subscribed to all event actions for the event source. ",
+										Validators: []validator.List{
+											listvalidator.UniqueValues(),
+										},
```

Comment:
We can't add a validation because these values are not controlled by the webhook service but by the product service. If we add a validation we limit the resource to existing integrations. New events created by product services that have individual action, like packer has `revoke`, won't be compatible with the resource until a new version is released with the change.

Reply:
Ah fair point. You can mark this comment as resolved then.